diff --git a/gimpopenvino/plugins/openvino_utils/tools/openvino_common/models_ov/fastsd/paths.py b/gimpopenvino/plugins/openvino_utils/tools/openvino_common/models_ov/fastsd/paths.py
index 97f9af8..0d3c13b 100644
--- a/gimpopenvino/plugins/openvino_utils/tools/openvino_common/models_ov/fastsd/paths.py
+++ b/gimpopenvino/plugins/openvino_utils/tools/openvino_common/models_ov/fastsd/paths.py
@@ -3,6 +3,7 @@ import constants
 from pathlib import Path
 from time import time
 from utils import get_image_file_extension
+from gimpopenvino.plugins.openvino_utils.tools.tools_utils import base_model_dir
 
 
 def join_paths(
@@ -17,7 +18,7 @@ def get_file_name(file_path: str) -> str:
 
 
 def get_app_path() -> str:
-    app_dir = os.path.dirname(__file__)
+    app_dir = base_model_dir
     work_dir = os.path.dirname(app_dir)
     return work_dir
 
